### ISSUE TITLE
[Feature Request] Password Tag

### ISSUE THREAD

**Issue by faelBrunnoS** [SpikeHimself/XPortal#58]
The portal now has two TAG fields:

- **NAMETAG**: The default tag for portals in the game.
- **PASSTAG**: A tag that works like a password.

If the destination portal has a `PASSTAG`, then the origin portal must also have the same `PASSTAG` to be able to teleport to it.

![image](https://github.com/SpikeHimself/XPortal/assets/83606488/802fc3d7-5369-4fb9-adf1-b45f2ce6beb7)

**Comment by SokBojowy** [comment]
Big YES! for Password Tag. I created my account just to support thia post. Great and very needed idea.

**Comment by SpikeHimself** [comment]
Hi, thank you for your support, and sorry for the extremely late response. Life's been very busy.


> Password TAG

I have to give this a bit more time in my head. I'm unsure if adding a password is feasible, as you'd need a field to set a password as well as one where you enter it when selecting a new destination, and it will quickly get clunky. Also this would have to be an optional component to the UI, as not everyone will want this, so I have to redo some of the UI code for that too.

> Remove Distance

The next XPortal update will include a config option `HidePortalDistance` which allows you to remove the distance indicator from the list of portals.

> Show List on Enter

By 'Enter' do you mean entering the portal (i.e. walking into it)? I think that will quickly get annoying as most of the time you'd just want to travel. Maybe I misunderstood.

**Comment by faelBrunnoS** [comment]
Yes, my suggestion is to show the list when you enter the portal. I understand that it can be annoying when you just want to travel! That's why I suggested it as an "option", so you can activate or deactivate it in the options, that way those who want the option active (like me) could just activate it!

I completely understand about the password, as it would be a new system and you would have to change the hud. But it was just an idea and it would be interesting to have it in the mod, since there is no portal mod with this, it would be a totally unique mod. I hope you give this idea some thought :)

**Comment by SpikeHimself** [comment]
> Yes, my suggestion is to show the list when you enter the portal. I understand that it can be annoying when you just want to travel! That's why I suggested it as an "option", so you can activate or deactivate it in the options, that way those who want the option active (like me) could just activate it!

Does this include portals that already have a destination configured? Or only for portals that don't have a connection yet?

If it is only for portals that aren't connected yet, then it is the same request as #63, which I think might be a nice addition.

If it's for all portals (connected or otherwise), I fear that, even though you say you do want this, you will inevitably end up annoyed by it, and so it's a lot of effort for nothing.

Additionally, with the request in #42 going live soon, you'll be able to immediately connect portals upon building them, which might clash with what you're asking.

**Comment by faelBrunnoS** [comment]
I think it could appear on all portals. But let's say there is a "Default Portal" feature in the mod, so when you enter the portal, before showing the list the mod checks if the "Default" checkbox is checked, if it is checked then it doesn't show the list and goes straight to the default portal.

In short, this option to show the list when you enter is for when you are in a portal that can go to several destinations, for example, my base I can go to a mine, or a mountain, or a boss. That way, I don't have to click on the list every time I travel and just enter the portal.

**Comment by faelBrunnoS** [comment]
I understand about the Default Portal feature.
The 2 really can't work at the same time, unless "Default" was different from just checking a checkbox.
If "Default" when active forces all portals to travel to it, then "Show on Enter" is not compatible.

There are only 2 ways for them to be compatible:
1 - There is another Default check, something like "Go To Default" active by default (not feasible).
2 - There are both options in the configuration to be activated, only one of which can be activated at the same time.

I can only see these solutions!

**Comment by faelBrunnoS** [comment]
I understand how Default works, I just explained it briefly. So keep thinking that those are the only options.

**Comment by faelBrunnoS** [comment]
So if "Default" only serves to define the connection between portals when the portal is built, then I don't see any compatibility problem. Because the player will only activate Show on Enter if he wants the benefits of this function.

**Comment by Jester2020** [comment]
something similar to this would also solve my issue of simply wanting to have some portal be private and not show in the list.
